### DISCUSSION TITLE
[Need help] Web browser history

### DISCUSSION THREAD

**Discussion by Nbzh** [arkivanov/Decompose#805] [Q&A]
Hello,

I'm starting a (very simple) projet  to learn to develop applications with Compose Multiplatform. I'm using "Decompose" for the navigation but I have an issue, so I'm requesting help to underdand what I'm doing wrong.

I'm trying to handle the Web browser history. I can set a path depending on the "Config" model but I cannot get the Config from the path  (I guess): 

When starting the projet on the browser, i have {HOST}/articles and then navigate to {HOST}/articles/{articleId}. The actions back / forward are working but I cannot refresh the page ; I get an error 

```
Cannot GET /articles
```
I followed this tutorial : https://arkivanov.github.io/Decompose/navigation/stack/browser-history/

A difference, I don't have a "JS app" but "Wasm App". I don't know if there are things to change.

I set those params

```
init {
        webHistoryController?.attach(
            navigator = nav,
            serializer = Config.serializer(),
            stack = _stack,
            getPath = ::getPathForConfig,
            getConfiguration = ::getConfigForPath,
        )
    }

private fun getPathForConfig(config: Config): String =
        when (config) {
            is Config.List -> "articles"
            is Config.Detail -> "/articles/${config.article}"
        }

 private fun getConfigForPath(path: String): Config =
        if (path.startsWith("/articles/") && path.length > 10)
            Config.Detail(path.substring(10))
        else Config.List
```

I added the dev server config (devServerConfig.js)  in the folder wasmJsMain, maybe it's not the right place.

You can find the code there : https://github.com/Nbzh/kmp_test_shop/blob/develop/composeApp/src/commonMain/kotlin/bzh/nvdev/melishop/data/NavigationModels.kt

Thank you for any help.

**Comment by arkivanov** [comment]
If you are registered in Kotlinlang Slack, then you can read about this [here](https://kotlinlang.slack.com/archives/C01F2HV7868/p1719747783647939?thread_ts=1719740202.360719&cid=C01F2HV7868).

TLDR: 

> You can try adding `<base href="/">` to your `index.html` `<head></head>` section.

**Reply by Nbzh** [reply]
Hello, thank you for the answer. 
Unfortunatly it didn't work.

**Reply by arkivanov** [reply]
Sad to hear that didn't help. I recommend asking in the Kotlinlang Slack, as this issue is not related to Decompose. Thanks!
